conversant/prompt_chatbot.py

```diff
@@ -244,56 +244,76 @@ def generate_prompt_update_examples(self, query: str) -> str:
-        """Generates (partial) reply to a query given a chat history.
+    def append_to_chat_history(
+        self, query: str, response: str, current_prompt: str, add_new_elem: bool
+    ) -> None:
+        """Appends the (partial) reply to chat history.
 
         Args:
-            query (str): A query passed to the prompt chatbot.
-            is_from_scratch (bool): Tells the chatbot if the reply
-            should be generated from scratch or it is the continuation
-            of the previous chunk
-
-        Returns:
-            Interaction: Dictionary of query and generated LLM response
-            bool: Indicates if the generation should be stopped
+            query (str):  A query passed to the prompt chatbot.
+            response (str): Response of the chatbot to query
+            current_prompt (str): Current prompt to save to prompt history
+            add_new_elem (bool): Indicates if a new element should be added to
+            chat history or if the last element should be modified only
         """
-        current_prompt = self.generate_prompt_update_examples(query)
-
-        generated_object = self.co.generate(
-            model=self.client_config["model"],
-            prompt=current_prompt,
-            max_tokens=TOKENS_PER_REQUEST,
-            temperature=self.client_config["temperature"],
-            frequency_penalty=self.client_config["frequency_penalty"],
-            presence_penalty=self.client_config["presence_penalty"],
-            stop_sequences=self.client_config["stop_sequences"],
-        )
-
-        response = generated_object.generations[0].text
-        is_final_chunk = False
-
-        stop_seq = self.should_stop(response)
-        if stop_seq != None or response == "":
-            is_final_chunk = True
-            if stop_seq != None:
-                response = response[: -len(stop_seq)]
-
-        if is_from_scratch:
-            response = response.lstrip()
+        if add_new_elem:
             self.chat_history.append(self.prompt.create_interaction(query, response))
             self.prompt_size_history.append(
                 self.co.tokenize(
                     self.prompt.create_interaction_string(query, response)
                 ).length
             )
             self.prompt_history.append(current_prompt)
-
         elif response != "":
             self.chat_history[-1]["bot"] += response
             self.prompt_history[-1] += response
             self.prompt_size_history[-1] += self.co.tokenize(response).length
 
-        return self.chat_history[-1]["bot"], is_final_chunk
+    def partial_reply(self, query: str) -> str:
+        """Generates (partial) reply to a query given a chat history.
+
+        Args:
+            query (str): A query passed to the prompt chatbot.
+            is_from_scratch (bool): Tells the chatbot if the reply
+            should be generated from scratch or it is the continuation
+            of the previous chunk
+
+        Returns:
```

Comment:
Let's also add a 
```
Yields:
```
section of the docstring (just like in the [guide](https://google.github.io/styleguide/pyguide.html#doc-function-returns))